### ISSUE TITLE
Loxone integration stops working

### ISSUE THREAD

**Issue by arjenvrh** [JoDehli/PyLoxone#176]
**Which Model do you use from loxone? Miniserver? Gen1 or Gen2?**
Gen1

**Which software version does your loxone use?**
14.0.3.28

**How did you install HomeAssistant? Over Hassio or manual install?**
Proxmox mit Hassio

**Which Version do you use of HomeAssistant?**
2023.4.6

**Describe the bug**
The integration works for about a day, after which no more communication between Loxone and Homeassistant is taking place. Everything was still working before March (and has been working stable for the last few years, thanks!). However, as I upgraded the Loxone firmware and Home Assistant in this period, I am unsure what the last stable configuration was. 

Note that the problem also occured with PyLoxone 0.4 (I tried the 0.5 betas as well). I recently updated Loxone to 14.0, however, the problem also occured in 13.x. 

**Paste the error log with the following settings:**

[home-assistant_2023-04-25T04-52-20.642Z.log](https://github.com/JoDehli/PyLoxone/files/11319033/home-assistant_2023-04-25T04-52-20.642Z.log)

**Comment by JoDehli** [comment]
@arjenvrh the log seams form the beta version. Please go back to the latest stable release if you do not miss some features. I am not really sure if the beta will become the next stable release. The beta is a complete rewrite of the websocket communication and  I am not really glad with the current implementation. It uses the  aiohttp lib for websocket and it is not really good for this.

I would use the latest stable release if you have problems. And now to your problem. I is interesting in your log is this line: 

2023-04-25 03:14:34.205 WARNING (MainThread) [homeassistant.components.unifi] Lost connection to UniFi Network

Can it be that you have lost the connection for some time? I know that there is a problem when sending a command like the keepalive message when the connection is not established at this time. Did you change something on you network? Because I did not changed anything which is related to connection for a long time. And on HA there has also nothing changed and on the loxone side also no big changes. Can you go back to the stable release and try again? 

I am not really glad how the implementation is working now but I have not so much time to rewrite it from scratch. The biggest problems I had in the past is with not stable connections and connection loses. 

And if you go back to the stable release I always had to install it twice over hacs. I do not know why but for downgrading I needed two installs.

**Comment by arjenvrh** [comment]
@JoDehli As stated I've been on the stable version a long time, I just upgraded to the beta this weekend to test if it had the same issue. I've also noticed the UniFi network warnings, which so far I've been unable to explain. The UniFi access points and controller have not had any updates over the last few months. 

However, as Home Assistant is on a wired network I do not expect this to be the problem. I'll try reverting to an older VM and version of Home assistant to see if this has any influence. I'll report back here in a few days with my findings.

**Comment by JoDehli** [comment]
@arjenvrh I now have another issue. I do not know if this is related to your problem. But I downgraded my installation to the 0.4.3 Version now. Lets see if any problem occur on my system. But as I already said I had problems to to downgrade over hacs from the beta release. I had to delete the custom components folder manually and then install the 0.4.3 from hacs to get the version running.

**Comment by JoDehli** [comment]
@arjenvrh Ok I tried to reproduce the error and I think I found something. It is definitely related to the stability of the network. I have tried hundreds of times to reproduce the error. I managed to reproduce the error only 1 or 2 times on my system. I was not able to force the error in the 0.4.3 version. Therefore, the error should not actually occur in 0.4.3. I have now also created a new beta version where I caught the error. You are welcome to try the latest version. 

As I said, I haven't quite figured out when exactly the error occurs. I have cut the network connection and partly kept disconnected for several minutes and still the connection did not disconnect and I could still control my Loxone via Homeassistant. 


Translated with www.DeepL.com/Translator (free version)
